unify_mode_error.m:015: In clause for `foo(in, out)':
unify_mode_error.m:015:   in argument 2 of call to predicate `builtin.unify'/2:
unify_mode_error.m:015:   mode error: variable `Y' has instantiatedness `free',
unify_mode_error.m:015:   expected instantiatedness was `ground'.
